qt/widgets/common/inc/MantidQtWidgets/Common/DisplayType.h

```diff
@@ -0,0 +1,32 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2020 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "DllOption.h"
+
+/**
+ * Creates an object containing information about converting display coordinates
+ *to data coordinates within a workspace
+ **/
+
+namespace MantidQt {
+namespace MantidWidgets {
+
+class EXPORT_OPT_MANTIDQT_COMMON DisplayType {
+public:
+  DisplayType(){};
+  ~DisplayType(){};
+
+  virtual void convertToDataCoord(const double xDisplayCoord,
```

Comment:
I have changed this. I have had to use a vector rather than a tuple however as it needs to be exposed in the sip and I could not get it to work correctly.

Reply:
Can we use the %MethodCode trick again? Something like?
```c++
  SIP_PYOBJECT toDisplayCoord(double x, double y) const;
  %MethodCode
    auto cppTuple = sipCpp->toDisplayCoord(a0,a1);
    return boost::python::incref(boost::python::tuple(cppTuple).ptr());
  %End
```

Reply:
My main concern is to avoid the constant dynamic allocation of the vector in a function I assume is called a lot?